spec/models/user_api_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+RSpec.describe UserApi, type: :model do
+  it 'generates api key' do
+    user = create(:user)
+    params = {
+      first_name: user.first_name,
+      last_name: user.last_name,
+      email: user.email,
+      description: 'testing key generation',
+      url: 'http://test.com',
+      redirect_url: 'http://test.com'
+    }
+    user_api = UserApi.find_or_create_by(user_id: user.id)
+    user_api.update(params)
+
+    expect(UserApi.last.key.length).to eq(22)
+    expect(UserApi.last.user).to eq(user)
+    expect(UserApi.last.first_name).to eq(user.first_name)
+    expect(UserApi.last.last_name).to eq(user.last_name)
+    expect(UserApi.last.email).to eq(user.email)
+    expect(UserApi.last.description).to eq(params[:description])
+    expect(UserApi.last.url).to eq(params[:url])
```

Comment:
@rdavid1099 Each time we call `UserApi.last` we make a trip to the DB. Should probably assign to a variable and reference the variable in the tests. Should speed up this test a decent amount.